### PR TITLE
Development

### PR DESCRIPTION
**Pull request by julianeventeu**
##### What this PR made?
The version beta about our system, the PR contain just the sources approved by all team.

### CODE REVIEW

**Review by dariodinizg — APPROVED** [review]

File: calculator/models.py [hunk 1]
```diff
@@ -1,3 +1,143 @@
...
+from django.forms import DecimalField
+from django.core.validators import MinValueValidator
 
-# Create your models here.
+from decimal import Decimal, ROUND_DOWN
+from datetime import datetime, timezone
+
+
+def increment_loan_id():
+        id_loan = '{:015d}'.format(Loan.objects.count() + 1)
```
Comment (AlexandreProenca) [inline calculator/models.py:10]: I think this is a not smart solution, we can try minimize the database access.

File: calculator/models.py [hunk 2]
```diff
@@ -1,3 +1,143 @@
...
+class Client(models.Model):
+    """
+    Client Model
+    Defines the attributes of a client
+    """
+
+    name = models.CharField("Name", max_length=30)
+    surname = models.CharField("Last name", max_length=30)
+    email = models.EmailField("E-mail")
+    phone = models.BigIntegerField("Phone")
```
Comment (AlexandreProenca) [inline calculator/models.py:23]: Why BigIntegerField here?

File: calculator/models.py [hunk 3]
```diff
@@ -1,3 +1,143 @@
...
+    instalment = models.DecimalField(
+        "Instalment",
+        default=Decimal('0.0'),
+        max_digits=15,
+        decimal_places=2,
+        validators=[MinValueValidator(Decimal("0.01"))],
+    )
+
+    def _instalment_adjustment(self):
+        loans_history = self.client.loan_set.all()
```
Comment (AlexandreProenca) [inline calculator/models.py:77]: Every time that you execute this function you will bring all loans, if you have millions of user maybe it is a bad ideia. use a cache should be better.

File: calculator/models.py [hunk 4]
```diff
@@ -1,3 +1,143 @@
...
+        missed_payments = sum(
+            [
+                Payment.objects.filter(loan_id=loan, status="missed").count()
```
Comment (AlexandreProenca) [inline calculator/models.py:80]: Bad Ideia, always if possible we need minimize database access, we can get the last loan to decide if the loan will be allowed or denied.

**Review by AlexandreProenca — APPROVED** [review]

**Pull request merged**